File: VentaCupones.API/VentaCupones.BC/Modelos/Cupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VentaCupones.BC.Modelos
{
    public class Cupon
    {
        public int IDCupon { get; set; }

        public string Nombre { get; set; }

        public string Imagen { get; set; }

        public string Ubicacion { get; set;}

        public decimal PrecioCupon { get; set; }

        public string IDEmpresa {  get; set; }
        
        public string IDCategoria { get; set; }

        public decimal Descuento { get; set; }
    }
}

File: VentaCuponesPHP.API/Model/Cupon.php
<?php
class Cupon {
    public $IDCupon;
    public $Nombre;
    public $Imagen;
    public $Ubicacion;
    public $PrecioCupon;
    public $IDEmpresa;
    public $IDCategoria;
    public $Descuento;
    public $Habilitado;
    
    public function __construct($IDCupon, $Nombre, $Imagen, 
    $Ubicacion, $PrecioCupon, $IDEmpresa, $IDCategoria,
    $Descuento, $Habilitado) 
    {
        $this->IDCupon = $IDCupon;
        $this->Nombre = $Nombre;
        $this->Imagen = $Imagen;
        $this->Ubicacion = $Ubicacion;
        $this->PrecioCupon = floatval($PrecioCupon);
        $this->IDEmpresa = $IDEmpresa;
        $this->IDCategoria = $IDCategoria;
        $this->Descuento = floatval($Descuento);
        $this->Habilitado = $Habilitado;
    }

    public function toJSON() {
        return json_encode(get_object_vars($this));
    }
}
?>
